ldms/src/sampler/zfs_common/zfs_common.c

```diff
@@ -0,0 +1,137 @@
+
+/* -*- c-basic-offser: 8 -*- */
+/* Copyright 2022 Lawrence Livermore National Security, LLC
+ * See the top-level COPYING file for details.
+ *
+ * SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+ */
+
+#define _GNU_SOURCE
+/* Next we include the headers to bring in the zfslib in action */
+#include <stdlib.h>
+#include <ctype.h>
+#include <glob.h>
+#include <string.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <libzfs.h>
+#include <libzutil.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "ldms.h"
+#include "ldmsd.h"
+#include "config.h"
+#include "sampler_base.h"
+#include "zfs_common/zfs_common.h"
+#include <stddef.h>
+
+char *
```

Comment:
Please put the return type on the same line as the function name to match the rest of the LDMS code style (needs to be done in all files).